translator/translate/otel/processor/jmxprocessor/translator.go

```diff
@@ -0,0 +1,106 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package jmxprocessor
+
+import (
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/processor"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
+)
+
+const (
+	// Match types are in internal package from contrib
+	// Strict is the FilterType for filtering by exact string matches.
+	regexp = "regexp"
+)
+
+var (
+	jmxKey           = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey)
+	activeMqKey      = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.ActiveMqKey)
+	cassandraKey     = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.CassandraKey)
+	hbaseKey         = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.HbaseKey)
+	hadoopKey        = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.HadoopKey)
+	jettyKey         = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.JettyKey)
+	jvmKey           = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.JvmKey)
+	kafkaKey         = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.KafkaKey)
+	kafkaConsumerKey = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.KafkaConsumerKey)
+	kafkaProducerKey = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.KafkaProducerKey)
+	solrKey          = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.SolrKey)
+	tomcatKey        = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.TomcatKey)
+	wildflyKey       = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.WildflyKey)
```

Comment:
nit: All of these can be `common.ConfigKey(jmxKey, common.ActiveMqKey)` (and all the other combinations) to make it clearer and a bit easier to read.

Reply:
Yeah, nice catch on this